src/nvim/ex_cmds.c

```diff
@@ -5836,3 +5867,328 @@ void set_context_in_sign_cmd(expand_T *xp, char_u *arg)
     }
   }
 }
+
+/// return the size of a long in term of digit
+int width_long(long nb) {
```

Comment:
Isn't it the `log10()` function you are trying to implement ? Seemes to be a core C function